vscode-dotnet-runtime-library/src/Utils/WebRequestWorkerSingleton.ts

```diff
@@ -110,10 +108,68 @@ export class WebRequestWorker
 
         this.client = setupCache(uncachedAxiosClient,
             {
-                storage: buildMemoryStorage(),
-                ttl: this.cacheTimeToLive
+                storage: buildMemoryStorage(
+                    false,
+                ),
             }
         );
+
+        // This function must be a member of 'this' so it correctly 'binds' to the instance of the class and has access to class members in a different scope environment
+        this.cacheCleanupRunner = function ()
+        {
+            if (this?.client?.storage !== null && this?.client?.storage !== undefined)
+            {
+                try
+                {
+                    if ((process.hrtime.bigint() - this.lastCacheCleanTimeNs) >= this.cacheTtlMs * 1e6)
+                    {
+                        // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+                        (this.client.storage as any)?.clear();
+                        this.lastCacheCleanTimeNs = process.hrtime.bigint();
+                    }
+                }
+                catch (err)
```

Comment:
Can you be more precise with these errors then? Not sure if that's a thing in ts

Reply:
The thrown object can be any type, while we could write handlers for specific error types, the action we'd take would generally be the same. I think thats a good question though. There is likely a better way to do this and I remember briefly looking into it and not finding much, though. You can at least try to assume it will throw an Error type but that is not necessarily a requirement. If any dependency in the code chain breaks the throw error convention then it would fail unexpectedly here 🤔